packages/ai-chat-ui/src/browser/chat-view-language-contribution.ts

```diff
@@ -164,33 +206,42 @@ export class ChatViewLanguageContribution implements FrontendApplicationContribu
                 }
             }
         }
-        return { suggestions };
-    }
 
-    provideToolCompletions(model: monaco.editor.ITextModel, position: monaco.Position): ProviderResult<monaco.languages.CompletionList> {
-        return this.getSuggestions(
-            model,
-            position,
-            PromptText.FUNCTION_CHAR,
-            this.toolInvocationRegistry.getAllFunctions(),
-            monaco.languages.CompletionItemKind.Function,
-            tool => tool.id,
-            tool => tool.name,
-            tool => tool.description ?? ''
-        );
+        return { suggestions };
     }
 
     protected async triggerVariableArgumentPicker(): Promise<void> {
         const inputEditor = monaco.editor.getEditors().find(editor => editor.hasTextFocus());
         if (!inputEditor) {
             return;
         }
+
         const model = inputEditor.getModel();
         const position = inputEditor.getPosition();
         if (!model || !position) {
             return;
         }
 
+        // Get the current line's content and the word at cursor
+        const lineContent = model.getLineContent(position.lineNumber);
+        const wordInfo = model.getWordUntilPosition(position);
+
+        // Only trigger if current word starts with #, followed by actual text
+        if (!wordInfo.word.startsWith(PromptText.VARIABLE_CHAR) || wordInfo.word.length <= 1) {
```

Comment:
This prevents the regression no 3 I [mentioned before](https://github.com/eclipse-theia/theia/pull/15030#pullrequestreview-2644624680). The word is actually empty, because we end with a `:` if the variable has mandatory arguments. So we would actually need to check if the character at the position is `PromptText.VARIABLE_SEPARATOR_CHAR` (see code below on line 246 etc.).